### ISSUE TITLE
Generate other CI/CD config files?

### ISSUE THREAD

**Issue by arschles** [luckyframework/lucky#1077]
Hi folks, I just started using Lucky today. This thing looks awesome! The first thing I noticed is that it generates a skeleton Travis CI config file (`.travis.yml`). I was wondering if you'd be open to having it generate other files as well? If so, I'm happy to submit PRs to have it generate Drone CI and GitHub actions.

Feel free to let me know, and if you're not ok with it, not a problem in the world 😄

**Comment by jwoertink** [comment]
Hey @arschles! Welcome to Lucky 🎉  

This file is actually generated by crystal https://github.com/crystal-lang/crystal/blob/8934d4e91bd7a47d3dabca82fd002522dadca18a/src/compiler/crystal/tools/init.cr#L298 I expect it's because travis gives you free CI when your project is open source. Lucky just generates a standard crystal app, then fills in the extra bits around it. I think having built-in github actions would be neat, or at least an easy tutorial on how to add it in. Maybe we could just make some documentation on our website on how to switch that out? What do you think about that?

**Comment by paulcsmith** [comment]
Yeah Travis was originally because crystal had it so we modified it. Travis has since gone downhill a bit. I’d love to have a GitHub actions file built in and have docs for others in the website. Most everyone uses GitHub and being able to just push your repo and have it work would be amazing!!

**Comment by paulcsmith** [comment]
And thanks for the kind words! Welcome to Lucky 🍀

**Comment by arschles** [comment]
Thanks @jwoertink and @paulcsmith. I'm new to crystal and lucky. I really appreciate the framework and how welcoming you all are.

I had no idea the travis file is generated from crystal. I assumed it was the lucky CLI! I'm cool with writing some documentation for how to add it for now. Should it go in the lucky docs?

**Comment by paulcsmith** [comment]
@arschles We're glad yo have you! And yes some guides would be really cool. Maybe in the new testing section! The gist is that you would duplicate one of these other guides: https://github.com/luckyframework/website/blob/master/src/actions/guides/testing/introduction.cr and then add the guide to the sidebar links https://github.com/luckyframework/website/blob/a8a8005635f13602b4cec4fb1eb7f9a6af0f763e/src/models/guides_list.cr#L60-L65. If you wanted to do that it'd be awesome! Even a basic draft would be 💯 

And just to clarify the travis.yml is generated by Crystal, but we modify it heavily to make it work with Lucky out of the box. I'm totally ok switching to GitHub actions once we get a working version. I haven't spent any time on that, but hopefully one of these days we can do it!
